physics/module_mp_thompson.F90

```diff
@@ -1419,8 +1422,13 @@ SUBROUTINE mp_gt_driver(qv, qc, qr, qi, qs, qg, ni, nr, nc,     &
                nifa1d(k) = nifa(i,k,j)
             enddo
          else
+            lsml = lsm(i,j)
             do k = kts, kte
-               nc1d(k) = Nt_c/rho(k)
+               if(lsml == 0) then
```

Comment:
I brought up this point in a telecon.  This might be sub-optimal.  It is relatively clear that the value zero represents ocean, but as a model user for nearly 30 years, how does the flag value "sea" represent something different? What about inland lakes, where do they reside in this schema?  I ask because aerosol differences over inland lakes are rather inconsequential and I was informed that lake points are going to receive the aerosol value for land, not ocean but this does not seem convincing to given a 3-category schema.

Reply:
land mask value 0 includes inland lakes. Ice is only over water (ocean and lake).  So the best modification would be:
               if(lsml == 1) then
                 nc1d(k) = Nt_c_l/rho(k)
               else
                 nc1d(k) = Nt_c_o/rho(k)
               endif

Reply:
That is what was used in the GFS high-reso and coupled experiments.  Thanks @ChunxiZhang-NOAA.   I agree.

Reply:
Thanks Ruiyu, I've modified those land mask related changes.